api-proposal.md

```diff
@@ -1,90 +1,152 @@
-# Proposal for ecosystem website api.
+# Proposal for Ecosystem Website API
 
 ## API's
 
 ---
 
-### Package List
+### Error handling
 
-We could have some specifc string matching for categories and tags in the query if thats what the user is searching for to maybe speed things up. `cat:<category>` and `tag:<string>`.
+On error, each endpoint returns a HTTP status code 500, and the following body
+```javascript
+data: {
+  error: string // containing an error message to be displayed in the UI
+}
+```
 
-- GET `/api/v1/packages?query=:string&page=:number`
-  ```
-  items: [<package>]
-  ```
+### Authentication
 
----
+An authenticated user is authenticated against GitHub.
+An authenticated administrator is defined by a set of GitHub usernames in the configuration of the backend.
+
+### Endpoints
 
-### Package
+#### Package
 
 - **Schema**
 
-  ```
+  ```javascript
   <package> {
-    id: number
-    name: string
-    description: string
-    readme: string // maybe from github or something?
-    image: blob || string
+    id: string // a unique, immutable string identifiying this package. allowed: [a-z-_]{2,}
+    name: string // human readable name
+    description: string 
+    readme: string // user can enter a markdown description
+    image: blob // we should restrict the image size to 5kb
```

Comment:
We should just get these from github instead of hosting them ourselves.

Reply:
Agreed & addressed.

Reply:
Actually, I mixed something up: There is no image for a project in GitHub. I always thought about individual users here, but this is about projects.

In that case, I fear we need to go back using a blob in the DB?

Reply:
Oh thats true, yeah I guess it's fine.  I've never really dealt with images stored in the database, we always uploaded them somewhere and put the path in the DB.

But I don't see any real problems with it.

Reply:
Should be doable. We might need an endpoint to retrieve the image.